pkg/resourcegenerator/networking/network_policy.go

```diff
@@ -160,6 +160,27 @@ func getIngressRules(opts NetPolOpts) []networkingv1.NetworkPolicyIngressRule {
 
 	// Allow grafana-agent to scrape
 	if opts.IstioEnabled {
+		promScrapeRuleAlloy := networkingv1.NetworkPolicyIngressRule{
+			From: []networkingv1.NetworkPolicyPeer{
+				{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"kubernetes.io/metadata.name": "grafana-alloy"},
+					},
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app.kubernetes.io/instance": "alloy",
+							"app.kubernetes.io/name":     "alloy",
+						},
+					},
+				},
+			},
+			Ports: []networkingv1.NetworkPolicyPort{
+				{
+					Port: util.PointTo(util.IstioMetricsPortName),
```

Comment:
This is the port named "istio-metrics". Does it exist on Alloy today? I only saw one called "http-metrics"

Reply:
This is the port that should be _scraped by_ alloy, e.g. the Istio sidecar.